src/media-segment-request.js

```diff
@@ -0,0 +1,281 @@
+import videojs from 'video.js';
+import { createTransferableMessage } from './bin-utils';
+
+export const REQUEST_ERRORS = {
+  ABORTED: -101,
+  TIMEOUT: -102,
+  FAILURE: 2
+};
+
+/**
+ * Turns segment byterange into a string suitable for use in
+ * HTTP Range requests
+ */
+const byterangeStr = function(byterange) {
+  let byterangeStart;
+  let byterangeEnd;
+
+  // `byterangeEnd` is one less than `offset + length` because the HTTP range
+  // header uses inclusive ranges
+  byterangeEnd = byterange.offset + byterange.length - 1;
+  byterangeStart = byterange.offset;
+  return 'bytes=' + byterangeStart + '-' + byterangeEnd;
+};
+
+/**
+ * Defines headers for use in the xhr request for a particular segment.
+ */
+const segmentXhrHeaders = function(segment) {
+  let headers = {};
+
+  if (segment.byterange) {
+    headers.Range = byterangeStr(segment.byterange);
+  }
+  return headers;
+};
+
+const abortAll = (activeXhrs) => {
+  Object.keys(activeXhrs).forEach((xhrKey) => {
+    const xhr = activeXhrs[xhrKey];
+
+    // only abort xhrs that haven't had a response
+    if (!xhr.responseTime) {
+      // set an aborted property so that we can correctly
+      // track the request and not treat it as an error
+      xhr.aborted = true;
+      xhr.abort();
+    }
+  });
+};
+
+const getRequestStats = (request) => {
+  return {
+    bandwidth: request.bandwidth,
+    bytesReceived: request.bytesReceived || 0,
+    roundTripTime: request.roundTripTime || 0
+  };
+};
+
+const getProgressStats = (progressEvent) => {
+  const request = progressEvent.target;
+  const roundTripTime = Date.now() - request.requestTime;
+  const stats = {
+    bandwidth: Infinity,
+    bytesReceived: 0,
+    roundTripTime: Math.max(roundTripTime, 1)
+  };
+
+  if (progressEvent.lengthComputable) {
+    stats.bytesReceived = progressEvent.loaded;
+    stats.bandwidth = (stats.bytesReceived / roundTripTime) * 8 * 1000;
```

Comment:
I think this should use `stats.roundTripTime` or use `Math.max` when initializing `roundTripTime` to avoid division by 0 (I assume the use of `Math.max` above was to not report a round trip of 0)